sdk/eventhub/Azure.Messaging.EventHubs/assets/live-tests-azure-setup.ps1

```diff
@@ -56,139 +56,12 @@ Import-Module Az.Resources
-  Write-Host "$($indent)`t`t`t`tprincipal must not already exist."
-  Write-Host ""
-
-  Write-Host "$($indent)-AzureRegion`t`tThe Azure region that resources should be created in.  This value should be"
-  Write-Host "$($indent)the name of the region, in lowercase, with no spaces.  For example: southcentralus"
-  Write-Host ""
-  Write-Host "$($indent)`t`t`t`tDefault: South Central US (southcentralus)"
-  Write-Host ""  
-}
-
-function SelectRandomCharacters
-{    
-    <#
-      .SYNOPSIS
-        Selects a number of random characters from a set.
-    #>
-
-    param
-    (
-        [Parameter(Mandatory=$true)]
-        [int]$length, 
-
-        [Parameter(Mandatory=$true)]
-        [string]$characters
-    ) 
-
-    $random = (1..$length | ForEach { Get-Random -Maximum $characters.length })
-    return (-join $characters[$random])
-}
-
-function GenerateRandomPassword()
-{
-    <#
-      .SYNOPSIS
-        Generates a random password that can be assigned to a service principal.
-        
-      .DESCRIPTION
-        The password generated by this function will contain a mix of alpha,
-        numeric, and special characters and will vary a small amount in its length.        
-    #>
-
-    $baseLength = (Get-Random -Minimum 28 -Maximum 37)      
-    $upper = (Get-Random -Minimum 6 -Maximum ([int][Math]::Ceiling($baseLength / 3)))  
-    $special = (Get-Random -Minimum 2 -Maximum 3)     
-    $lower = ($baseLength - $upper - $special - $special)                                 
-
-    $password = SelectRandomCharacters $lower "abcdefghiklmnoprstuvwxyz"
-    $password += SelectRandomCharacters $upper "ABCDEFGHKLMNOPRSTUVWXYZ"
-    $password += SelectRandomCharacters $special "1234567890"
-    $password += SelectRandomCharacters $special "!$%&/()=?}][{@#*+"
-
-    $scrambled = ($password.ToCharArray()) | Get-Random -Count ($password.Length)
-    return (-join $scrambled)
-}
-
-function TearDownResources
-{
-    <#
-      .SYNOPSIS
-        Cleans up any Azure resources created by the script.
-        
-      .DESCRIPTION
-        Responsible for cleaning up any Azure resources created 
-        by the script in case of failure.
-    #>
-    
-    param
-    (
-        [Parameter(Mandatory=$true)]
-        [bool] $cleanResourceGroup
-    )
-    
-    Write-Host("Cleaning up resources that were created:")
-    
-    if ($cleanResourceGroup)
-    {
-        try 
-        {
-            Write-Host "`t...Removing resource group `"$($ResourceGroupName)`""
-            Remove-AzResourceGroup -Name "$($ResourceGroupName)" -Force | Out-Null
-        }
-        catch 
-        {
-            Write-Error "The resource group: $($ResourceGroupName) could not be removed.  You will need to delete this manually."
-            Write-Error ""            
-            Write-Error $_.Exception.Message
-        }
-    }
-}
+. .\functions\validation-functions.ps1

```

Comment:
The organization of these is nice, though I do wonder if we're going a bit too granular.    Not that I expect people to really drill in to the "functions" folder, but if they did, would finding this many scripts be intimidating?   The counterpoint would be "would someone trying to debug be more intimidated by one fat common script with a bunch of loosely-related things?"

I'm honestly not sure.

Reply:
I am on the fence (and a copycat :P)

I broke the script down as it was becoming to big. However I can put them back in a single script and use regions?